lib/ansible/module_utils/basic.py

```diff
@@ -1754,7 +1754,15 @@ def _check_argument_values(self, spec=None, param=None):
                             if len(overlap) == 1:
                                 (param[k],) = overlap
 
-                        if param[k] not in choices:
+                        if isinstance(param[k], SEQUENCETYPE):
+                            for item in param[k]:
+                                if item not in choices:
+                                    choices_str = ",".join([to_native(c) for c in choices])
+                                    msg = "values of %s must be one of: %s, got: %s" % (k, choices_str, item)
+                                    if self._options_context:
+                                        msg += " found in %s" % " -> ".join(self._options_context)
+                                    self.fail_json(msg=msg)
+                        elif param[k] not in choices:
```

Comment:
We could cut down on the iterations by using sets to determine if an option is not in the choices.

```
invalid = set(param[k]).difference(choices)
if invalid:
    # create message using results of invalid
```

```
>>> set(['alpha', 'omega']).difference([ 'alpha', 'beta', 'gamma' ])
set(['omega'])
```

Reply:
Good idea. Although performance wise it's unclear to me what's more efficient.
Readability is more important.